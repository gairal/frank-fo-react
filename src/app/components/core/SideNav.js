import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { withStyles } from '@material-ui/core/styles';
import {
  AppBar,
  Toolbar,
  Drawer,
  List,
  ListItem,
  ListItemText,
  ListItemIcon,
  Icon,
  Hidden,
  Divider,
  Typography,
} from '@material-ui/core';

const drawerWidth = 240;
const styles = () => ({
  drawerPaper: {
    width: drawerWidth,
  },
});

class SideNav extends Component {
  static propTypes = {
    routes: PropTypes.arrayOf(PropTypes.shape()).isRequired,
    classes: PropTypes.shape().isRequired,
    mobileOpen: PropTypes.bool.isRequired,
    toggleDrawer: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);

    const { routes, toggleDrawer } = this.props;

    this.drawer = (
      <div>
        <AppBar position="static" color="primary">
          <Toolbar>
            <Typography variant="title" component="h1" color="inherit">
              frank g.
            </Typography>
          </Toolbar>
        </AppBar>
        <List>
          {routes.map(e => (
            <NavLink
              to={e.path}
              key={e.key}
              onClick={toggleDrawer}
              className="nav-link"
            >
              <ListItem button>
                <ListItemIcon>
                  <Icon color="error">{e.icon}</Icon>
                </ListItemIcon>
                <ListItemText primary={e.key} />
              </ListItem>
            </NavLink>
          ))}
          <Divider />
          <ListItem button>
            <ListItemText primary="version" secondary="4.2.0-react" />
          </ListItem>
        </List>
      </div>
    );
  }

  render() {
    const { classes, mobileOpen, toggleDrawer } = this.props;

    return (
      <React.Fragment>
        <Hidden mdUp>
          <Drawer
            variant="temporary"
            open={mobileOpen}
            onClose={toggleDrawer}
            ModalProps={{
              keepMounted: true, // Better open performance on mobile.
            }}
            classes={{
              paper: classes.drawerPaper,
            }}
          >
            {this.drawer}
          </Drawer>
        </Hidden>
        <Hidden smDown implementation="css">
          <Drawer
            variant="permanent"
            open
            classes={{
              paper: classes.drawerPaper,
            }}
          >
            {this.drawer}
          </Drawer>
        </Hidden>
      </React.Fragment>
    );
  }
}

export default withStyles(styles)(SideNav);
