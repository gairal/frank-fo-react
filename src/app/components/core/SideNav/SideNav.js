import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  AppBar,
  Toolbar,
  Drawer,
  List,
  ListItem,
  ListItemText,
  ListItemIcon,
  Icon,
  Hidden,
  Typography } from 'material-ui';

export default class SideNav extends Component {
  static propTypes = {
    routes: PropTypes.arrayOf(PropTypes.object).isRequired,
  }

  constructor(props) {
    super(props);
    this.state = {
      mobileOpen: false,
    };

    this.drawer = (
      <div>
        <AppBar position="static" color="primary">
          <Toolbar>
            <Typography type="headline" component="h1" color="inherit">
              frank g.
            </Typography>
          </Toolbar>
        </AppBar>
        <List>
          {this.props.routes.map(e => (
            <ListItem button key={e.path}>
              <ListItemIcon>
                <Icon color="contrast">{e.icon}</Icon>
              </ListItemIcon>
              <ListItemText primary={e.path} />
            </ListItem>
          ))}
          <ListItem button>
            <ListItemText primary="version" secondary="4.0.0-react" />
          </ListItem>
        </List>
      </div>
    );
  }

  render() {
    return (
      <div>
        <Hidden mdUp>
          <Drawer
            type="temporary"
            open={this.state.mobileOpen}
            onRequestClose={this.handleDrawerToggle}
            ModalProps={{
              keepMounted: true, // Better open performance on mobile.
            }}
          >
            {this.drawer}
          </Drawer>
        </Hidden>
        <Hidden mdDown implementation="css">
          <Drawer
            type="permanent"
            open
          >
            {this.drawer}
          </Drawer>
        </Hidden>
      </div>
    );
  }
}
