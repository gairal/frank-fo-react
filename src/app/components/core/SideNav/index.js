import sideNav from './SideNav';

export default sideNav;
