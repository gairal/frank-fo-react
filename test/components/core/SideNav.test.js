/* eslint-disable function-paren-newline,comma-dangle  */
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SideNav from '@/components/core/SideNav';

describe('components > core > SideNav', () => {
  it('Renders correctly default', () => {
    const component = renderer.create(
      <BrowserRouter>
        <SideNav
          mobileOpen
          toggleDrawer={jest.fn}
          routes={
            [
              {
                key: 'skills',
                icon: 'build',
                path: 'categories',
              },
            ]
          }
        />
      </BrowserRouter>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
