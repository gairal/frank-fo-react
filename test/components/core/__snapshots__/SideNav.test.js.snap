// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > core > SideNav Renders correctly default 1`] = `
<div
  className="HiddenCss-smDown-7"
>
  <div
    className="MuiDrawer-docked-17"
  >
    <div
      className="MuiPaper-root-28 MuiPaper-elevation0-30 MuiDrawer-paper-18 SideNav-drawerPaper-1 MuiDrawer-paperAnchorLeft-19 MuiDrawer-paperAnchorDockedLeft-23"
    >
      <div>
        <header
          className="MuiPaper-root-28 MuiPaper-elevation4-34 MuiAppBar-root-55 MuiAppBar-positionStatic-59 MuiAppBar-colorPrimary-61"
        >
          <div
            className="MuiToolbar-root-63 MuiToolbar-gutters-64"
          >
            <h1
              className="MuiTypography-root-65 MuiTypography-title-71 MuiTypography-colorInherit-84"
            >
              frank g.
            </h1>
          </div>
        </header>
        <ul
          className="MuiList-root-89 MuiList-padding-90"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="categories"
            onClick={[Function]}
          >
            <div
              className="MuiButtonBase-root-103 MuiListItem-root-93 MuiListItem-default-96 MuiListItem-gutters-100 MuiListItem-button-101"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-107 MuiIcon-colorError-111 MuiListItemIcon-root-106"
              >
                build
              </span>
              <div
                className="MuiListItemText-root-113"
              >
                <span
                  className="MuiTypography-root-65 MuiTypography-subheading-72 MuiListItemText-primary-116"
                >
                  skills
                </span>
              </div>
              <span
                className="MuiTouchRipple-root-119"
              />
            </div>
          </a>
          <hr
            className="MuiDivider-root-126"
          />
          <div
            className="MuiButtonBase-root-103 MuiListItem-root-93 MuiListItem-default-96 MuiListItem-gutters-100 MuiListItem-button-101"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="MuiListItemText-root-113"
            >
              <span
                className="MuiTypography-root-65 MuiTypography-subheading-72 MuiListItemText-primary-116"
              >
                version
              </span>
              <p
                className="MuiTypography-root-65 MuiTypography-body1-74 MuiTypography-colorTextSecondary-87 MuiListItemText-secondary-117"
              >
                4.1.0-react
              </p>
            </div>
            <span
              className="MuiTouchRipple-root-119"
            />
          </div>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
